### PR TITLE
Corrected check for libtiff feature

### PR DESCRIPTION
**Pull request by radarhere**
https://github.com/python-pillow/Pillow/blob/db21e7de1cd56ec798f3866ffc132b6f4119b572/Tests/test_tiff_ifdrational.py#L55-L61

This is the wrong way around. `WRITE_LIBTIFF` should be `True` when libtiff is available.

To demonstrate, if I skip [building libtiff on Windows](https://github.com/radarhere/Pillow/commit/c217c6900569baf28206e87430a1b896a12863b1), it [fails](https://github.com/radarhere/Pillow/actions/runs/8673645602). With [this change](https://github.com/radarhere/Pillow/commit/94f7d1ff3915b0160c81159eabcb8021f37e720c), it [passes.](https://github.com/radarhere/Pillow/actions/runs/8673821109)

### CODE REVIEW

**Comment by nulano** [comment]
Should we use `pytest.parameterize` with a skip mark to make it clear that the `WRITE_LIBTIFF = True` test was skipped in the logs?

**Comment by radarhere** [comment]
Ok, I've updated the commit.

File: Tests/test_tiff_ifdrational.py [hunk 1]
```diff
@@ -52,18 +54,17 @@ def test_nonetype() -> None:
...
-    methods = [True]
-    if features.check("libtiff"):
-        methods.append(False)
-
-    for libtiff in methods:
-        TiffImagePlugin.WRITE_LIBTIFF = libtiff
+@pytest.mark.parametrize(
+    "libtiff", (pytest.param(True, marks=skip_unless_feature("libtiff")), False)
+)
+def test_ifd_rational_save(tmp_path: Path, libtiff: bool) -> None:
```
Comment (nulano) [inline Tests/test_tiff_ifdrational.py:60]:
```suggestion
def test_ifd_rational_save(monkeypatch: pytest.MonkeyPatch, tmp_path: Path, libtiff: bool) -> None:
```

File: Tests/test_tiff_ifdrational.py [hunk 2]
```diff
@@ -52,18 +54,17 @@ def test_nonetype() -> None:
...
+    im = hopper()
+    out = str(tmp_path / "temp.tiff")
+    res = IFDRational(301, 1)
 
-        im = hopper()
-        out = str(tmp_path / "temp.tiff")
-        res = IFDRational(301, 1)
-        im.save(out, dpi=(res, res), compression="raw")
+    TiffImagePlugin.WRITE_LIBTIFF = libtiff
```
Comment (nulano) [inline Tests/test_tiff_ifdrational.py:65]:
```suggestion
    monkeypatch.setattr(TiffImagePlugin, "WRITE_LIBTIFF", libtiff)
```

File: Tests/test_tiff_ifdrational.py [hunk 3]
```diff
@@ -52,18 +54,17 @@ def test_nonetype() -> None:
...
+    im.save(out, dpi=(res, res), compression="raw")
+    TiffImagePlugin.WRITE_LIBTIFF = False
```
Comment (nulano) [inline Tests/test_tiff_ifdrational.py:67]:
```suggestion
```

**Review by nulano — COMMENTED** [review]
Do we want to use monkeypatch in case the save method raises an exception?

**Comment by radarhere** [comment]
I've pushed a commit to use monkeypatch whenever the tests set `READ_LIBTIFF` or `WRITE_LIBTIFF`

File: Tests/test_file_libtiff.py [hunk 4]
```diff
@@ -756,10 +747,9 @@ def check_write(libtiff: bool) -> None:
             with Image.open(out) as reloaded:
                 assert icc_profile == reloaded.info["icc_profile"]
 
-        libtiffs = []
+        libtiffs = [False]
         if Image.core.libtiff_support_custom_tags:
             libtiffs.append(True)
```
Comment (nulano) [inline Tests/test_file_libtiff.py:752]: This can also use `pytest.mark.parametrize`

Comment (radarhere) [reply Tests/test_file_libtiff.py]: Done

File: Tests/test_imagesequence.py [hunk 5]
```diff
@@ -60,10 +60,9 @@ def test_tiff() -> None:
 
 
 @skip_unless_feature("libtiff")
-def test_libtiff() -> None:
-    TiffImagePlugin.READ_LIBTIFF = True
+def test_libtiff(monkeypatch: pytest.MonkeyPatch) -> None:
+    monkeypatch.setattr(TiffImagePlugin, "READ_LIBTIFF", True)
     _test_multipage_tiff()
```
Comment (nulano) [inline Tests/test_imagesequence.py:65]: While not as useful, these two tests can also be combined into one with `pytest.mark.parametrize` for slightly shorter code.

**Review by nulano — APPROVED** [review]
Looks good; I just noticed one more test that can be parametrized.

**Review by nulano — APPROVED** [review]

**Pull request merged**